server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -848,10 +848,9 @@ private static String getIgnoreAboveDefaultValue(final Settings settings) {
         Property.Final
     );
 
-    public static final boolean SYNTHETIC_VECTORS = new FeatureFlag("mapping_synthetic_vectors").isEnabled();
-    public static final Setting<Boolean> INDEX_MAPPING_SOURCE_SYNTHETIC_VECTORS_SETTING = Setting.boolSetting(
-        "index.mapping.synthetic_vectors",
-        false,
+    public static final Setting<Boolean> INDEX_MAPPING_EXCLUDE_SOURCE_VECTORS_SETTING = Setting.boolSetting(
+        "index.mapping.exclude_source_vectors",
```

Comment:
We have a lot of repetition of the string "index.mapping.exclude_source_vectors", we should think about parameterizing it.